Lib/calendar.py

```diff
@@ -660,7 +658,7 @@ def timegm(tuple):
     return seconds
 
 
-def main(args):
+def main(args=None):
```

Comment:
Can you explain the changes to `main()` -- whilst I believe they should have the same behaviour, changing this should ideally be done in another PR.

A

Reply:
The alternative approach would be to hijack `sys.argv` either via monkeypatching or restoring the value at the end, but that is messy. Can you keep `main(sys.argv)`?

A

Reply:
> Can you keep `main(sys.argv)`?

I'm afraid I can't without an additional conditional code. However, I think `main(sys.argv[1:])` might work. Can you explain why it's better to keep the old way?

I think my change is compliant with the `argparse` protocol and also backwards compatible (it uses `sys.argv` by default).

See:
https://github.com/python/cpython/blob/30610d28374f5a9698d456cebf3ae496ac01af51/Lib/argparse.py#L1861-L1874

Reply:
> Can you explain why it's better to keep the old way?

Easier to review a PR that does less, especially if it touches less 'real' stdlib code. But as you point out it seems it'd have to change anyway so might as well just call `main()`.

A